test/Proxy.js

```diff
@@ -0,0 +1,23 @@
+const { assert } = require('chai');
+const { ethers, upgrades } = require('hardhat');
+let proxy;
+let GaslessV3;
+describe('Proxy', function () {
+    before(async () => {
+        GaslessV3 = await ethers.getContractFactory('GaslessV3');
+        proxy = await upgrades.deployProxy(GaslessV3, [
+            hre.network.config.config.wrappedTokenAddress,
+            hre.network.config.config.gasForSwap,
+            hre.network.config.config.gasForApproval,
+            hre.network.config.config.defaultGasPrice,
+        ]);
+    });
+
+    it('works after upgrading', async function () {
```

Comment:
i don't think this test is actually checking what we need to check

ideally, we want to update create a proxy contract with implementation A. then we need to change the implementation to B. then we need to change gasForSwap and check if the value changed according to implementation B and not A